docs/source/actor-rhel7-o-rhel8.md

```diff
@@ -0,0 +1,321 @@
+
+But, if you want to execute it manually, by tests purposes, you can also use snactor for it. First we need to make sure that all messages that will be consumed are generated and stored. For this example, this means running _OSReleaseCollector_ actor with the _--save-output_ option of snactor:
+
+```shell
+# snactor run --verbose --save-output OSReleaseCollector
+2018-11-23 13:06:30.706 INFO     PID: 17996 leapp: Logging has been initialized
+2018-11-23 13:06:30.753 INFO     PID: 17996 leapp.repository.system_upgrade_el7toel8: A new repository 'system_upgrade_el7toel8' is initialized at /usr/share/leapp-repository/repositories/system_upgrade/el7toel8
+2018-11-23 13:06:30.803 INFO     PID: 17996 leapp.repository.common: A new repository 'common' is initialized at /usr/share/leapp-repository/repositories/common
+```
+
+Now, you can execute _CheckOSRelease_ actor and verify that it consumes the previously generated message and produces a message saying that the target system is not supported by Leapp upgrade process. You don’t need to specify which message will be consumed, snactor will take care of it.
+
+```shell
+# snactor run --verbose --print-output CheckOSRelease
+2018-11-23 13:11:15.549 INFO     PID: 18126 leapp: Logging has been initialized
+2018-11-23 13:11:15.578 INFO     PID: 18126 leapp.repository.system_upgrade_el7toel8: A new repository 'system_upgrade_el7toel8' is initialized at /usr/share/leapp-repository/repositories/system_upgrade/el7toel8
+2018-11-23 13:11:15.617 INFO     PID: 18126 leapp.repository.common: A new repository 'common' is initialized at /usr/share/leapp-repository/repositories/common
+[
+  {
+    "stamp": "2019-04-30T13:12:05.706317Z",
+    "hostname": "leapp-20190429150826",
+    "actor": "check_os_release",
+    "topic": "report_topic",
+    "context": "0ac49430-1b29-4940-92bb-3e81da85f8af",
+    "phase": "NON-WORKFLOW-EXECUTION",
+    "message": {
+      "hash": "686355c648832f5399cffc5bd0e7e8312d2b77577cdeec94f674a2565090cf7f",
+      "data": "{\"audience\": [\"sysadmin\"], \"detail\": \"{\\\"summary\\\": \\\"Minimal supported OS version for upgrade process: 7.6\\\"}\", \"flags\": [\"inhibitor\"], \"renderers\": {\"html\": \"<h2 class=\\\"report-title\\\">{{ title }}</h2><p class=\\\"report-summary\\\">{{ summary }}</p>\", \"plaintext\": \"{{ title }}\\n{{ summary }}\"}, \"severity\": \"medium\", \"title\": \"Unsupported OS version\"}"
+    },
+    "type": "Report"
+  }
+]
+```
+
+To flush all saved messages from the repository database, run `snactor messages clear`.
+
+## Writing tests for an actor
+
+[Read the tutorial](https://leapp.readthedocs.io/en/latest/unit-testing.html) for writing an running unit and component tests
+
+## Best practices
+
+Read the best practices [document](https://github.com/oamg/leapp-repository/blob/master/docs/best-practices.md)
```

Comment:
Please add a link to the general 'python guidelines' document https://github.com/oamg/leapp-guidelines/blob/master/python-coding-guidelines.md
We should merge the best practices and guidelines one day as well.